bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractComponentImpl.java

```diff
@@ -122,7 +123,7 @@ protected void setCurrentPage(Page currentPage) {
     @PostConstruct
     private void init() {
         // Setting currentPage to ResourcePage to prevent id miss-match when invoked via iframe mode in sites.
-        if (currentPage != null && resource != null) {
+        if (currentPage != null && resource != null && request.getAttribute("embeddedAdaptiveForm") != null) {
```

Comment:
This is on public space, why is this attribute required ? How will it impact forms developer, is this something which needs to be documented or internal ?

Reply:
Even if required rename the constant to REQ_ATTR_EMBEDDED_ADAPTIVEFORM

Reply:
This was for a bug fix, where there are duplicate ids generated in case form container is directly dragged into the site page versus when form is embedded as div (not iframe). This will be needed.
I will change it to a constant.